sessiontoken.go

```diff
@@ -20,7 +20,9 @@ func (c *Client) CreateSessionToken(tokenRequest *sessiontoken.CreateTokenReques
 		return nil, err
 	}
 
-	resp, err := c.doRequest("POST", SessionTokenAPIURL, nil, bytes.NewReader(payload))
+	// we need to explicitly pass an empty token (which means it wont get set in the header)
+	// the API accepts either no token or a valid token, but not an empty token.
+	resp, err := c.doRequestWithToken("POST", SessionTokenAPIURL, nil, bytes.NewReader(payload), "")
```

Comment:
Hrm, what does a missing token mean in this context? The docs seem to imply this wouldn't work. I'll do some testing locally so I can catch up with you!

Reply:
Oh, I see. It's because the JSON body authenticates the user. Lemme finish testing it on my end then I'll review. Sorry for not knowing this already!

Reply:
My non-Go background desperately wanted to pass a nil here for the token but alas that’s not a possibility unless the function accepts it as a *string which adds unnecessary complexity IMO.

Reply:
Coulda fooled me, this looks pretty Go-ish to me!

Yeah, definitely avoid the pointer thing. The Go way for that would be to pass a `""`.